### PR TITLE
RabittMQ adapter

### PR DESCRIPTION
**Pull request by jfonseca-aneo**

### CODE REVIEW

File: Adaptors/RabbitMQ/tests/QueueStorageTests.cs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using Moq;
+
+using NUnit.Framework;
+
+using RabbitMQ.Client;
+
+namespace ArmoniK.Core.Adapters.RabbitMQ.Tests;
```
Comment (aneojgurhem) [inline Adaptors/RabbitMQ/tests/QueueStorageTests.cs:41]: Ces tests ont l'air d'être un copie de ceux d'AMQP. Est ce qu'on peut les mutualiser ?

File: Adaptors/Amqp/src/QueueMessageHandler.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public class QueueMessageHandler : IQueueMessageHandler
   }
 
   /// <inheritdoc />
-  public CancellationToken CancellationToken { get; }
+  public CancellationToken CancellationToken { get; set; }
```
Comment (aneojgurhem) [inline Adaptors/Amqp/src/QueueMessageHandler.cs:62]: Why set ?

File: Adaptors/RabbitMQ/src/PullQueueStorage.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  {
+    var nbPulledMessage = 0;
+
+    if (!IsInitialized)
+    {
+      throw new ArmoniKException($"{nameof(PullQueueStorage)} should be initialized before calling this method.");
+    }
+
+    // TODO: Fix this, I put it here cuz the method must be async to return an IAsyncEnumerable.
+    await Task.CompletedTask.ConfigureAwait(false);
```
Comment (aneojgurhem) [inline Adaptors/RabbitMQ/src/PullQueueStorage.cs:180]: Maybe try ValueTask that are lighter than Task. Or juste suppress the warning

File: Adaptors/RabbitMQ/src/PushQueueStorage.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    ConnectionRabbit.Channel!.ExchangeDeclare("ArmoniK.QueueExchange",
+                                              "direct");
+
+    var basicProperties = ConnectionRabbit.Channel!.CreateBasicProperties();
+    basicProperties.Priority = Convert.ToByte(priority);
+
+    foreach (var msg in messages)
+    {
+      ConnectionRabbit.Channel.BasicPublish("ArmoniK.QueueExchange",
```
Comment (aneojgurhem) [inline Adaptors/RabbitMQ/src/PushQueueStorage.cs:85]: No batch method ?

File: Adaptors/RabbitMQ/src/QueueMessageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      case QueueMessageStatus.Postponed:
+        /* Negative acknowledging this message will send it
+         to the retry exchange, see PullQueueStorage.cs */
+        channel_.BasicNack(deliverEvent_.DeliveryTag,
+                           false,
+                           false);
+        break;
+      case QueueMessageStatus.Failed:
+
+      case QueueMessageStatus.Processed:
```
Comment (aneojgurhem) [inline Adaptors/RabbitMQ/src/QueueMessageHandler.cs:86]: I would suggest a rework and documentation of those statuses

File: docker-compose/docker-compose.override.yml [hunk 6]
```diff
@@ -4,32 +4,46 @@ services:
...
+      - Components__QueueStorage=ArmoniK.Adapters.Amqp.QueueStorage
 
   armonik.control.metrics:
     environment:
       - ASPNETCORE_ENVIRONMENT=Development
 
   armonik.compute.pollingagent0:
+    depends_on:
+      queue:
+        condition: service_started
```
Comment (aneojgurhem) [inline docker-compose/docker-compose.override.yml:16]: Why in the override and not the regular docker-compose.yml ?
A bit of explanation for this should be better. Does it uses our health checks ?

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

File: Adaptors/Amqp/src/QueueMessageHandler.cs (hunk 2 above)
Comment (aneojgurhem) [reply Adaptors/Amqp/src/QueueMessageHandler.cs:62]: I don't remember the usage of the set part.

File: docker-compose/docker-compose.override.yml (hunk 6 above)
Comment (jfonseca-aneo) [reply docker-compose/docker-compose.override.yml:16]: It is already in the regular compose file. I forgot to remove it from here ^^. I'll add some doc.

File: Adaptors/RabbitMQ/src/PushQueueStorage.cs (hunk 4 above)
Comment (jfonseca-aneo) [reply Adaptors/RabbitMQ/src/PushQueueStorage.cs:85]: The .net client does not provide with an implementation, but in their git repository they show how to do it in [this example](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/TestApplications/MassPublish/Program.cs). We can try that out, but in a separate PR I'd say

File: Adaptors/RabbitMQ/src/PullQueueStorage.cs (hunk 3 above)
Comment (jfonseca-aneo) [reply Adaptors/RabbitMQ/src/PullQueueStorage.cs:180]: Ok; I'll give it a try

File: Adaptors/RabbitMQ/src/PushQueueStorage.cs (hunk 4 above)
Comment (aneojgurhem) [reply Adaptors/RabbitMQ/src/PushQueueStorage.cs:85]: ok

File: Adaptors/Amqp/src/QueueMessageHandler.cs (hunk 2 above)
Comment (jfonseca-aneo) [reply Adaptors/Amqp/src/QueueMessageHandler.cs:62]: I put the set to pass the right cancellation token in the RabbitMQ adapter, its only used there.

File: Adaptors/RabbitMQ/src/QueueMessageHandler.cs (hunk 5 above)
Comment (jfonseca-aneo) [reply Adaptors/RabbitMQ/src/QueueMessageHandler.cs:86]: Agree... but in another PR

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**